### PR TITLE
feat!: Update to ACVM 0.21.0

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Problem\*

Resolves 

## Summary\*

Migrates to ACVM 0.21.0 (Stepwise ACVM)

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: crates/noirc_driver/src/lib.rs [hunk 1]
```diff
@@ -305,5 +305,10 @@ pub fn compile_no_check(
     let (circuit, debug, abi) =
         create_circuit(program, options.show_ssa, options.show_brillig, show_output)?;
 
+    if options.print_acir {
+        println!("Unoptimised ACIR for main:");
```
Comment (kevaundray) [inline crates/noirc_driver/src/lib.rs:309]:
```suggestion
        println!("Unoptimized ACIR for main:");
```

Comment (TomAFrench) [reply crates/noirc_driver/src/lib.rs:309]: Ah, thanks for this review. We actually have some issues with the unoptimized circuit being printed twice (leading to our wild goose chase for memop opcodes yesterday.)

**Comment by TomAFrench** [comment]
Actually, we need to hold off on merging until bberg uses the new serialisation format.

**Review by kobyhallx — APPROVED** [review]

**Comment by jfecher** [comment]
@kobyhallx are the bberg changes done?

**Comment by TomAFrench** [comment]
I've update bberg to use the new ACIR parser here: https://github.com/AztecProtocol/barretenberg/pull/631

As part of this I've had to remove the `bb-bin-tests` job as this uses legacy ssa artifacts which are incompatible with the new serialisation and the new SSA does not support recursion.

**Review by kobyhallx — APPROVED** [review]

**Pull request merged**